cmake/public/cuda.cmake

```diff
@@ -214,9 +210,19 @@ endif()
 
 # nvToolsExt
 add_library(torch::nvtoolsext INTERFACE IMPORTED)
-set_property(
-    TARGET torch::nvtoolsext PROPERTY INTERFACE_LINK_LIBRARIES
-    CUDA::nvToolsExt)
+find_path(
+  nvtx3_dir
+  NAMES nvtx3
+  PATHS ${CUDA_INCLUDE_DIRS}
+  NO_DEFAULT_PATH)
+find_package_handle_standard_args(nvtx3 DEFAULT_MSG nvtx3_dir)
+if(NOT nvtx3_FOUND)
+  set(nvtx3_dir "${CMAKE_CURRENT_LIST_DIR}/../../third_party/NVTX/c/include")
+  message(WARNING "use NVTX library in ${nvtx3_dir}")
+endif()
+include_directories(SYSTEM "${nvtx3_dir}")
+target_include_directories(torch::nvtoolsext INTERFACE "${nvtx3_dir}")
```

Comment:
Is the `include_directories` command necessary? If so, it suggests a something is not linking to `torch::nvtoolsext` that should be.

Reply:
It is not necessary now since the minimal CUDA version supported is CUDA 11.8